src/installers.ts

```diff
@@ -849,13 +849,141 @@ export const installJsonMod: VortexWrappedInstallFunc = (
 
   return Promise.resolve({ instructions });
 };
+
+// INI (includes Reshade?)
+export const testForIniMod: VortexWrappedTestSupportedFunc = (
+  api: VortexAPI,
+  log: VortexLogFunc,
+  files: string[],
+  _gameId: string,
+): Promise<VortexTestResult> => {
+  // Make sure we're able to support this mod.
+  const correctGame = _gameId === GAME_ID;
+  log("info", "Checking for INI files: ", _gameId);
+  if (!correctGame) {
+    // no mods?
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  const filtered = files.filter(
+    (file: string) => path.extname(file).toLowerCase() === INI_MOD_EXT,
+  );
+
+  if (filtered.length === 0) {
+    log("info", "No INI files.");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+
+  if (
+    files.some(
+      (file: string) =>
+        path.basename(file).includes(CET_MOD_CANONICAL_INIT_FILE) ||
+        path.extname(file) === REDS_MOD_CANONICAL_EXTENSION,
+    )
+  ) {
+    log("info", "INI file detected within a CET or Redscript mod, aborting");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  if (files.includes(CET_GLOBAL_INI)) {
+    log("info", "CET Installer detected, not processing as INI");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  return Promise.resolve({
+    supported: true,
+    requiredFiles: [],
+  });
+};
+
+export const installIniMod: VortexWrappedInstallFunc = (
+  api: VortexAPI,
+  log: VortexLogFunc,
+  files: string[],
+  _destinationPath: string,
+): Promise<VortexInstallResult> => {
+  // This installer gets called for both reshade and "normal" ini mods
+  const archiveName = path.basename(_destinationPath, ".installing");
+
+  const filtered = files.filter(
+    (file: string) => path.extname(file) === INI_MOD_EXT,
+  );
+  const shaderFiles = files.filter(
+    (file: string) => file.includes(SHADERS_DIR) && !file.endsWith(path.sep),
+  );
+
+  let reshade = false;
+  // We're going to make a reasonable assumption here that reshades will
+  // only have reshade ini's, so we only need to check the first one
+  const data: string = readFileSync(
```

Comment:
Minor: this could be an `= await readFileAsync` to free up the loop.

Reply:
And let’s figure out a way to test file reading since it’s new functionality! If it’s a bigger effort, then for now could only allow Reshades that are in `bin\x64\`?

Reply:
Ok, using fs.readFile lets me catch an error reading the file _and_ read async;

Reply:
Unfortunately it breaks things...  Reshade files get handled properly but regular files don't.  I'm sure I'm missing something

Reply:
Oh, right, `readFileAsync` is the old callback API, `readFile` is the Promise API so that's the right one.. try-catch around the `await` _should_ work (and we don't want to move to using Promises directly) but if it's causing problems just leave it sync for now :)